### ISSUE TITLE
netlifyctl deploy crashes when output is not a TTY

### ISSUE THREAD

**Issue by malcolmr** [netlify/netlifyctl#65]
I'm running netlifyctl headless, in a script executed on the end of a pipe. It looks like 0.2.0 crashes if the output isn't a TTY (?).

This is easy to demonstrate by running `netlifyctl deploy ... | cat`, which crashes with:

```
Counting objects .... 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x84ca01]

goroutine 1 [running]:
github.com/netlify/netlifyctl/operations.(*DeployObserver).OnSuccessfulStep(0xc4201526c0, 0xc4204eb5f0, 0xa, 0xc4204eb5f0)
	/home/david/src/github.com/netlify/netlifyctl/operations/deploy.go:43 +0xc1
github.com/netlify/netlifyctl/vendor/github.com/netlify/open-api/go/porcelain.walk.func1(0xc4204eb560, 0x22, 0x10559e0, 0xc420287ba0, 0x0, 0x0, 0x0, 0x0)
	/home/david/src/github.com/netlify/netlifyctl/vendor/github.com/netlify/open-api/go/porcelain/deploy.go:480 +0x343
path/filepath.walk(0xc4204eb560, 0x22, 0x10559e0, 0xc420287ba0, 0xc4204eb4a0, 0x0, 0x0)
	/usr/local/go/src/path/filepath/path.go:351 +0x81
path/filepath.walk(0x7ffd72860e0f, 0x17, 0x10559e0, 0xc420287a00, 0xc4204eb4a0, 0x0, 0x30)
	/usr/local/go/src/path/filepath/path.go:376 +0x414
path/filepath.Walk(0x7ffd72860e0f, 0x17, 0xc4204eb4a0, 0x0, 0xc4204eb470)
	/usr/local/go/src/path/filepath/path.go:398 +0x14c
github.com/netlify/netlifyctl/vendor/github.com/netlify/open-api/go/porcelain.walk(0x7ffd72860e0f, 0x17, 0x1056920, 0xc4201526c0, 0x0, 0x0, 0xc4200c1080)
	/home/david/src/github.com/netlify/netlifyctl/vendor/github.com/netlify/open-api/go/porcelain/deploy.go:487 +0x1be
github.com/netlify/netlifyctl/vendor/github.com/netlify/open-api/go/porcelain.(*Netlify).DoDeploy(0xc4202845e0, 0x7fa822de02c0, 0xc4204eae70, 0xc4200ccf30, 0x0, 0xba29a0, 0xbc3d00, 0xc4204f2900)
	/home/david/src/github.com/netlify/netlifyctl/vendor/github.com/netlify/open-api/go/porcelain/deploy.go:173 +0xc5
github.com/netlify/netlifyctl/vendor/github.com/netlify/open-api/go/porcelain.(*Netlify).DeploySite(0xc4202845e0, 0x7fa822de02c0, 0xc4204eae70, 0xc4204eb050, 0x24, 0x7ffd72860e0f, 0x17, 0x0, 0x0, 0x0, ...)
	/home/david/src/github.com/netlify/netlifyctl/vendor/github.com/netlify/open-api/go/porcelain/deploy.go:153 +0x9d
github.com/netlify/netlifyctl/commands/deploy.(*deployCmd).deploySite(0xc4202f66c0, 0x7fa822de0280, 0xc4204eae70, 0xc420282000, 0xc4201f6280, 0x0, 0x8, 0xc4203638b0, 0x84f912)
	/home/david/src/github.com/netlify/netlifyctl/commands/deploy/deploy.go:85 +0x61b
github.com/netlify/netlifyctl/commands/deploy.(*deployCmd).(github.com/netlify/netlifyctl/commands/deploy.deploySite)-fm(0x7fa822de0280, 0xc4204eae70, 0xc420282000, 0xc4201f6280, 0x0, 0x8, 0x0, 0x10a3da0)
	/home/david/src/github.com/netlify/netlifyctl/commands/deploy/deploy.go:44 +0x69
github.com/netlify/netlifyctl/commands/middleware.DebugMiddleware.func1(0x7fa822de0280, 0xc4204eae70, 0xc420282000, 0xc4201f6280, 0x0, 0x8, 0x0, 0x0)
	/home/david/src/github.com/netlify/netlifyctl/commands/middleware/middleware.go:87 +0x509
github.com/netlify/netlifyctl/commands/middleware.LoggingMiddleware.func1(0x7fa822de0280, 0xc4204eae10, 0xc420282000, 0xc4201f6280, 0x0, 0x8, 0x10534a0, 0xc420016320)
	/home/david/src/github.com/netlify/netlifyctl/commands/middleware/middleware.go:113 +0x1f7
github.com/netlify/netlifyctl/commands/middleware.AuthMiddleware.func1(0x7fa822de0280, 0xc4204ead80, 0xc420282000, 0xc4201f6280, 0x0, 0x8, 0x1, 0xc4204eacc0)
	/home/david/src/github.com/netlify/netlifyctl/commands/middleware/middleware.go:124 +0x1d4
github.com/netlify/netlifyctl/commands/middleware.ClientMiddleware.func1(0x7fa822de0200, 0xc420016320, 0xc420282000, 0xc4201f6280, 0x0, 0x8, 0xc687bd, 0x4)
	/home/david/src/github.com/netlify/netlifyctl/commands/middleware/middleware.go:177 +0x316
github.com/netlify/netlifyctl/commands/middleware.NewRunFunc.func1(0xc420282000, 0xc4201f6280, 0x0, 0x8, 0x0, 0x0)
	/home/david/src/github.com/netlify/netlifyctl/commands/middleware/middleware.go:52 +0xf5
github.com/netlify/netlifyctl/vendor/github.com/spf13/cobra.(*Command).execute(0xc420282000, 0xc4201f6200, 0x8, 0x8, 0xc420282000, 0xc4201f6200)
	/home/david/src/github.com/netlify/netlifyctl/vendor/github.com/spf13/cobra/command.go:644 +0x3ef
github.com/netlify/netlifyctl/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1095c60, 0x10bf6b7, 0xc67fba, 0x3)
	/home/david/src/github.com/netlify/netlifyctl/vendor/github.com/spf13/cobra/command.go:734 +0x339
github.com/netlify/netlifyctl/commands.Execute()
	/home/david/src/github.com/netlify/netlifyctl/commands/netlify.go:41 +0x2a5
main.main()
	/home/david/src/github.com/netlify/netlifyctl/main.go:16 +0x7e
```

0.1.5 works fine, in comparison. (It doesn't have ac209cbf, which I suspect is the problem.)

**Comment by calavera** [comment]
Fixed with #67. I'll publish a new version in a few minutes.

**Comment by calavera** [comment]
I just published the version v0.2.1 that fixes this problem. Thanks a lot for reporting it 🙌
